src/kernel_density_estimation.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+/* @flow */
+
+var stddev = require('./sample_standard_deviation');
+
+var SQRT_2PI = Math.sqrt(2 * Math.PI);
+
+/**
+ * [Well-known kernels](https://en.wikipedia.org/wiki/Kernel_(statistics)#Kernel_functions_in_common_use)
+ * @private
+ */
+var kernels /*: {[string]: (number) => number} */ = {
+    /**
+     * The gaussian kernel.
+     * @private
+     */
+    gaussian: function (u) {
+        return Math.exp(-0.5 * u * u) / SQRT_2PI;
+    }
+};
+
+/**
+ * Well known bandwidth selection methods
+ * @private
+ */
+var bandwidthMethods /*: {[string]: (Array<number>) => number} */ = {
+    /**
+     * The commonly used variation of [Silverman's
+     * rule-of-thumb](https://en.wikipedia.org/wiki/Kernel_density_estimation#A_rule-of-thumb_bandwidth_estimator) found in Scott, D. W. (1992) Multivariate Density Estimation: Theory, Practice, and Visualization.
+     * @private
+     */
+    nrd: function (x /*: Array<number> */) {
```

Comment:
What is `nrd` an abbreviation of? I'm not seeing anything that refers to NRD in the linked Wikipedia page.

Reply:
Oops, yeah this comes from https://stat.ethz.ch/R-manual/R-devel/library/stats/html/bandwidth.html, and it's also used in `science.js`, but I've had no luck figuring out what it means.  What do you think is better:
 - Change the link from Wikipedia to the R documentation
 - Rename from `nrd` to `scott`, and add a link to the scipy docs that explicitly distinguish Silverman from Scott (they're basically the same modulo a constant, but the Wikipedia page doesn't get into it)
 - Just call it `silverman`, because it's searchable on the wikipedia page.

Reply:
Scipy link: https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.gaussian_kde.html

Reply:
"nrd" appears to stand for "Normal Reference Distribution", based on https://stat.ethz.ch/R-manual/R-devel/library/MASS/html/bandwidth.nrd.html.